### PR TITLE
Top headlines carousel

### PR DESCRIPTION
**Pull request by Diogo00dev**
# Short Description
*Created horizontal carousel for the topHeadlines*

# What changes in the code?
*Created HorizontalListTopHeadlines widget that contains a horizontal carousel for the topHeadlines
Created ArticleCard widget for the articles cards*

# Did you test?
*Yes*

### CODE REVIEW

File: flutter/news/lib/core/result.dart [hunk 1]
```diff
@@ -4,7 +4,7 @@ import 'package:news/core/error/failures.dart';
 part 'result.freezed.dart';
 
 @freezed
-abstract class Result<S> with _$Result<S> {
+class Result<S> with _$Result<S> {
```
Comment (juankysoriano) [inline flutter/news/lib/core/result.dart:7]: :+1:

File: flutter/news/lib/features/frontpage/presentation/widgets/horizontal_list_top_headlines.dart [hunk 2]
```diff
@@ -11,28 +10,58 @@ class HorizontalListTopHeadlines extends StatelessWidget {
...
+          const Align(
+            alignment: Alignment.centerLeft,
+            child: Padding(
+              padding: EdgeInsets.only(
+                left: 16.0,
+                top: 16.0,
+                bottom: 8.0,
+              ),
+              child: Text(
+                'TOP HEADLINES',
```
Comment (juankysoriano) [inline flutter/news/lib/features/frontpage/presentation/widgets/horizontal_list_top_headlines.dart:24]: on android we have a resources file with texts like this, what is the best practice for flutter? 

That's a good thing to investigate

File: flutter/news/lib/main.dart [hunk 3]
```diff
@@ -16,7 +16,7 @@ class MyApp extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return const MaterialApp(
-      title: 'asdsd',
+      title: 'Your News',
```
Comment (juankysoriano) [inline flutter/news/lib/main.dart:19]: long life to `asdsd`

File: flutter/news/test/widgets/article_card_test.dart [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import '../core/utils/extensions.dart';
+
+void main() {
+  final article = TopHeadlineViewState.from(
+    article: Stub.article(
+      title: "1",
+    ),
+  );
+
+  testWidgets(
```
Comment (juankysoriano) [inline flutter/news/test/widgets/article_card_test.dart:16]: any link or good resource that you know for learning more about widget testing?

Comment (Diogo00dev) [reply flutter/news/test/widgets/article_card_test.dart:16]: i guided myself by here https://docs.flutter.dev/cookbook/testing/widget/introduction

File: flutter/news/lib/features/frontpage/presentation/widgets/horizontal_list_top_headlines.dart (hunk 2 above)
Comment (Diogo00dev) [reply flutter/news/lib/features/frontpage/presentation/widgets/horizontal_list_top_headlines.dart:24]: actually there is no direct way to do this like in Android, i have found a couple of different ways to do it, maybe we should do a discussion about it

**Pull request merged**